src/services/blocks/trace/Trace.ts

```diff
@@ -0,0 +1,666 @@
+		const eventsByParentId = new Map<number, PEvent[]>();
+
+		for (const [idx, event] of this.traceBlock.events.entries()) {
+			const { key } = event.data.stringValues;
+			const keyPrefix = key?.slice(0, 64);
+			const storagePath = this.getStoragePathFromKey(keyPrefix);
+
+			const p = spansById.get(event.parentId);
+			const parentSpanId = p && {
+				name: p.name,
+				target: p.target,
+				id: p.id,
+			};
+
+			const parsed = { ...event, storagePath, parentSpanId, eventIndex: idx };
+
+			if (!eventsByParentId.has(parsed.parentId)) {
+				eventsByParentId.set(parsed.parentId, [parsed]);
+			} else {
+				eventsByParentId.get(parsed.parentId)?.push(parsed);
+			}
+
+			const extrinsicIndexOpt = this.maybeExtractIndex(parsed);
+			if (isSome(extrinsicIndexOpt)) {
+				extrinsicIndexBySpanId.set(parsed.parentId, extrinsicIndexOpt);
+			}
+		}
+
+		return { eventsByParentId, extrinsicIndexBySpanId };
+	}
+
+	/**
+	 * Create a mapping adress => Account events for that address based on an
+	 * array of all the events.
+	 *
+	 * This mapping is useful because we create operations based on consecutive
+	 * gets/puts to a single accounts balance data. So later on we can just take all
+	 * the account events from a single address and create `Operation`s from those.
+	 *
+	 * Note: Assumes passed in events are already sorted.
+	 *
+	 * @param events events with phase info
+	 * @returns
+	 */
+	accountEventsByAddress(events: PActionEvent[]): Map<string, PAccountEvent[]> {
+		return events.reduce((acc, cur) => {
+			if (
+				!(
+					(cur.storagePath as PalletKeyInfo).module == 'system' &&
+					(cur.storagePath as PalletKeyInfo).item == 'account'
+				)
+			) {
+				// filter out non system::account events
+				return acc;
+			}
+
+			const asAccount = this.toAccountEvent(cur);
+			const address = asAccount.address.toString();
+
+			if (!acc.has(address)) {
+				acc.set(address, [asAccount]);
+			} else {
+				acc.get(address)?.push(asAccount);
+			}
+
+			return acc;
+		}, new Map<string, PAccountEvent[]>());
+	}
+
+	/**
+	 * Convert an `PActionEvent` to a `PAccountEvent` by adding the decoded `accountInfo`
+	 * and `address`.
+	 *
+	 * Notes: will throw if event does not have system::account storagePath.
+	 *
+	 * @param event PActionEvent
+	 */
+	private toAccountEvent(event: PActionEvent): PAccountEvent {
+		const { storagePath } = event;
+		if (
+			!(
+				(storagePath as PalletKeyInfo).module == 'system' &&
+				(storagePath as PalletKeyInfo).item == 'account'
+			)
+		) {
+			throw new InternalServerError(
+				'Event did not have system::account path as expected'
+			);
+		}
+
+		// storage key = h(system) + h(account) + h(address) + address
+		//								^^^^^^^^^^^^^^^^^^^^storage item prefix
+		// Since this is a transparent key with the address at the end we can pull out
+		// the address from the key. Here we slice off the storage key + account hash.
+		const addressRaw = event.data.stringValues.key?.slice(96);
```

Comment:
Is the key size fixed, might be worth noting its size in the comment above if possible.